### PR TITLE
Compatibility with toml11 4.0

### PR DESCRIPTION
**Pull request by franzpoeschel**
toml11 4.0+ seems to be a breaking change. Especially, output formatting now works differently.

- [x] Add a toml11 4.0.3 CI run

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -278,7 +278,12 @@ if(openPMD_USE_INTERNAL_TOML11)
     add_subdirectory("${openPMD_SOURCE_DIR}/share/openPMD/thirdParty/toml11")
     message(STATUS "toml11: Using INTERNAL version '3.7.1'")
 else()
-    find_package(toml11 3.7.1 CONFIG REQUIRED)
+    # Since toml11 4.0 was a breaking change,
+    # it needs separate find_package() calls
+    find_package(toml11 3.7.0 CONFIG QUIET)
```
Comment (ax3l) [inline CMakeLists.txt]: I think we intentionally had 3.7.1 due to a bug fix in it:
```suggestion
    find_package(toml11 3.7.1 CONFIG QUIET)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -278,15 +278,13 @@ if(openPMD_USE_INTERNAL_TOML11)
...
-    # Need to check the version manually. toml11 4.0 was a breaking change, so
-    # find_package(toml11 3.7.1) would not accept versions 4.0+.
-    # We support both 3.* and 4.*.
-    if(${toml11_VERSION} VERSION_LESS 3.7.1)
-        message(FATAL_ERROR "toml11: Found version '${toml11_VERSION}', but requires version '3.7.1' at least.")
-    else()
-        message(STATUS "toml11: Found version '${toml11_VERSION}'")
+    # Since toml11 4.0 was a breaking change,
+    # it needs separate find_package() calls
+    find_package(toml11 3.7.0 CONFIG QUIET)
```
Comment (franzpoeschel) [inline CMakeLists.txt]: TODO: change the order here, prefer version 4 in doubt

File: .github/workflows/linux.yml [hunk 3]
```diff
@@ -133,6 +133,17 @@ jobs:
...
+        cd toml11
+        git checkout v4.0.3
+        mkdir build
+        cd build
+        cmake ..                                      \
+          -DCMAKE_INSTALL_PREFIX=../../toml11_install \
+          -DCMAKE_CXX_STANDARD_REQUIRED=OFF           \
+          -DCMAKE_CXX_STANDARD=11
+        make -j 2 install
+        cd ../..
```
Comment (ax3l) [inline .github/workflows/linux.yml]:
```suggestion
        git clone -b v4.0.3 https://github.com/ToruNiina/toml11
        cmake -S toml11 -B build_toml11               \
          -DCMAKE_INSTALL_PREFIX=../../toml11_install \
          -DCMAKE_CXX_STANDARD_REQUIRED=OFF           \
          -DCMAKE_CXX_STANDARD=11
        cmake --build build_toml11 -j 2 --target install
```

**Review by ax3l — APPROVED** [review]
Thank you :)

**Pull request merged**